nixos/modules/services/networking/cloudflare-warp.nix

```diff
@@ -0,0 +1,96 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.cloudflare-warp;
+in
+{
+  options.services.cloudflare-warp = {
+    enable = mkEnableOption "Cloudflare Zero Trust client daemon";
+
+    package = mkPackageOption pkgs "cloudflare-warp" { };
+
+    rootDir = mkOption {
+      type = types.str;
+      default = "/var/cloudflare-warp";
+      description = ''
+        Working directory for the warp-svc daemon.
+      '';
+    };
+
+    udpPort = mkOption {
+      type = types.port;
+      default = 2408;
+      description = ''
+        The UDP port to open in the firewall. Warp uses port 2408 by default, but fallback ports can be used
+        if that conflicts with another service. See the [firewall documentation](https://developers.cloudflare.com/cloudflare-one/connections/connect-devices/warp/deployment/firewall#warp-udp-ports)
+        for the pre-configured available fallback ports.
+      '';
+    };
+
+    openFirewall = mkOption {
```

Comment:
This could use `mkEnableOption` I believe. Additionally, it is discouraged to use `with lib;` in the whole module, but rather required functions/components should be called explicitly instead. like `lib.mkOption`.